crates/kitsune-captcha/src/mcaptcha.rs

```diff
@@ -0,0 +1,56 @@
+use async_trait::async_trait;
+use http::Request;
+use serde::{Deserialize, Serialize};
+use typed_builder::TypedBuilder;
+
+use crate::Result;
+use crate::{CaptchaBackend, ChallengeStatus};
+
+use kitsune_http_client::Client;
+
+#[derive(Clone, TypedBuilder)]
+pub struct Captcha {
+    pub widget_link: String,
+    pub verify_url: String,
+    pub secret_key: String,
+    pub site_key: String,
+    #[builder(default = Client::builder().build())]
+    client: Client,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct MCaptchaResponse {
+    valid: bool,
+}
+
+#[derive(Serialize, Deserialize, TypedBuilder)]
+struct Body {
+    token: String,
+    key: String,
+    secret: String,
+}
+
+#[async_trait]
+impl CaptchaBackend for Captcha {
+    async fn verify(&self, token: &str) -> Result<ChallengeStatus> {
+        let body = Body::builder()
+            .secret(self.secret_key.to_string())
+            .token(token.to_string())
+            .key(self.site_key.to_string())
+            .build();
+        let body = simd_json::to_string(&body)?;
+        let request = Request::post(self.verify_url.clone())
+            .header("Content-Type", "application/json")
+            .header("Accept", "application/json")
+            .body(body.into())?;
+        let response = self.client.execute(request).await?;
+        let mut response_bytes = response.text().await?.into_bytes();
```

Comment:
We do have a `.json()` method. The API is pretty close to `reqwest`'s

https://github.com/kitsune-soc/kitsune/blob/870f9688b7a297a85988141d79a06bdb322d3656/crates/kitsune-http-client/src/lib.rs#L369-L375